enhancements/sig-architecture/34-work-executor-group/README.md

```diff
@@ -0,0 +1,275 @@
+
+There can be different actors using the work api to distribute resource from
+the hub cluster. Currently the work agent has the cluster-admin access to the
+managed cluster which means that every hub users with write access to the
+work resources will be able to dispatch any resources to the managed cluster.
+This can be a security hole for OCM, e.g. one can apply/overwrite some
+cluster-scoped configuration to the managed cluster which potentially break/halt
+the cluster. Hence, we need a way to clarify the owner identity/role of the
+"ManifestWork" before it takes effect so that we can explicitly check whether 
+that owner has sufficient permission in the managed cluster.
+
+### Auditing the API request history from work agent
+
+Currently all the API requests raised by work agent in the managed cluster are
+from a fixed service account ("open-cluster-management-agent/klusterlet-work-sa"
+under the default installation). However in some cases, we need to clarify the
+source/owner identity of the `ManifestWork` resource so that we can record the
+api audits in a finer granularity. This will be helpful for the managed
+cluster's admin to be capable of knowing "who is operating that `ManifestWork`
+from the hub cluster".
+
+### Goals
+
+- `ManifestWork` resource be able to clarify the executor/owner identity before
+  actually writing the resources to the managed cluster.
+
+### Non-Goals
+
+- This proposal doesn't put any requirement to the selection of executor/owner in
+  the "ManifestWork", which allows hub admin to reference any valid Kubernetes 
+  user/role to be the executor/owner.
+  
+## Terms
+
+- `Executor Subject`: The subject which is used by the work agent when actually 
+  applying resources in the spoke cluster.
+
+
+## Function Spec
+
+1. A hub admin can limit the scope of the dispatch'able resources for a 
+   `ManifestWork` by setting a non-empty executor subject, so the work agent
+   in the spoke cluster can report failure when applying/modifying out-of-scope 
+   resources.
+2. A spoke admin can expand the permission scope of the executor subject 
+   (e.g. a service-account) by binding more ClusterRole/Role to it.
+3. (beta phase) A hub user should pass the authorization before writing
+   the resources in the work object. The hub cluster holds a policy configuration
+   of "which hub user has write access to which work resource" and a validating
+   admission webhook will handle the execution of that policy.
+
+## Proposal
+
+### API Changes
+
+This proposal adds a new field `executorSubject` to the `ManifestWork` api, 
+which is defaulted to nil if absent. A nil executor subject indicates that the 
+work agent doesn't perform any impersonation before requesting the hosting 
+managed cluster.
+
+
+A sample of the new `executorSubject` field in Yaml is shown below:
+
+```yaml
+apiVersion: work.open-cluster-management.io/v1
+kind: ManifestWork
+metadata: ...
+spec:
+  executorSubject:
```

Comment:
I think it would be better as

```yaml
spec:
  executor:
    subject:
      type: ServiceAccount
      ...
```

It give us space to add other field for executor in the future. But we could discuss this in the api PR.

Reply:
@qiujian16 updated, on the other hand i have a bit complex feeling about overly expanding the struct, which might involve complexity for the api consumers. but expanding this one may be fine, i can imagine we add other fields like `exectuor#rateLimit` or `executor#namespacedOnly` in the future